### PR TITLE
.coafile: Add ShellCheckBear

### PR DESCRIPTION
**Pull request by sangamcse**
This add a `bash` section and `bash` as default `shell`.
Modify .sh files using `shfmt -i 2 -ci` from https://github.com/mvdan/sh#shfmt.
This changes shell scripts according to Google's guide.

Closes https://github.com/coala/coala/issues/4952

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ dependencies:
     - sed -i '/source \/home\/ubuntu\/virtualenvs\//d' ~/.circlerc
     - nvm alias default node
     - npm install -g csslint
+    - sudo apt-get install shellcheck
```
Comment (jayvdb) [inline circle.yml]: This belongs in the other commit.

And I would expect that we also need to install shellcheck on Travis.  I am surprised it is already present on the travis VM.  I'll double check that in the next review.

p.s. I've created https://github.com/coala/coala-bears/issues/2317

Comment (sangamcse) [reply circle.yml]: Done :sweat_smile:

Comment (sangamcse) [reply circle.yml]: Also check https://travis-ci.org/coala/coala/jobs/350911877#L60.

Comment (jayvdb) [reply circle.yml]: We dont need a third commit.  Adding shellcheck to Circle Ci is part of enabling shellcheck on Circle CI.

File: .misc/deps.sh [hunk 2]
```diff
@@ -8,12 +8,12 @@ fi
 
 # Choose the python versions to install deps for
 case $CIRCLE_NODE_INDEX in
```
Comment (jayvdb) [inline .misc/deps.sh]: all of this circle stuff is going to get removed in the Circle CI patch, which is critical.
Lets get that in first.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5243/commits/c61314069164bf7ac4f1626d780383dc9ec96f6a https://github.com/coala/coala/pull/5243/commits/779b3ba67e5ca4341fc604ef897f3ba2db6da4ab

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5243/commits/779b3ba67e5ca4341fc604ef897f3ba2db6da4ab

File: .misc/deps.sh (hunk 2 above)
Comment (jayvdb) [reply .misc/deps.sh]: that is https://github.com/coala/coala/pull/5234

**Comment by sangamcse** [comment]
Ping @jayvdb, any chance of merging it?

**Comment by jayvdb** [comment]
Nah, @alphadose is close to finishing his one.  Push him to finish it :P

File: .misc/check_maintainership.sh [hunk 3]
```diff
@@ -8,6 +8,6 @@ source ../rultor_secrets.sh
 
 # Make sure the user is a maintainer
 if curl "https://api.github.com/teams/1238757/memberships/${author}?access_token=${GITHUB_TOKEN}" | grep -q "Not Found"; then
-    echo "@${author} is not in the maintainers group."
-    exit 1
+  echo "@${author} is not in the maintainers group."
```
Comment (jayvdb) [inline .misc/check_maintainership.sh]: see https://github.com/coala/coala/issues/5328#issuecomment-379592735

File: .misc/deps.sh (hunk 2 above)
Comment (jayvdb) [reply .misc/deps.sh]: merged.

**Comment by sangamcse** [comment]
Ping @jayvdb, Ready for your reviews. 🤘

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5243/commits/dba361cfd66960cfd5eb3e9acde33c579901037d https://github.com/coala/coala/pull/5243/commits/17db7895b1e55877b9c2de94e1e8f6fde7f2e586 https://github.com/coala/coala/pull/5243/commits/9e6f36588915d5d0cc3f80a83f44a83635d01af2

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**